### PR TITLE
Removed gammapy dependency

### PR DESCRIPTION
**Pull request by eneights**
Altered detector response to use astromodels instead of gammapy, and changed format of config file to be compatible with astromodels

### CODE REVIEW

**Review by israelmcmc — CHANGES_REQUESTED** [review]
Thanks, @eneights! I haven't run all this yet, but it seems like it should accomplish the intended changes. Most of these comments are about style.

One general comment, before I forget: once we get this running, the integration code should be part of 3ML/astromodels instead.

**Comment by israelmcmc** [comment]
See also https://github.com/cositools/intro_cosipy/pull/2

File: docs/tutorials/DetectorResponse.ipynb [hunk 1]
```diff
@@ -621,34 +937,47 @@
...
    ],
    "source": [
-    "spectrum = PowerLawSpectralModel(index = 3,\n",
-    "                                 amplitude = 10**-3 / u.cm / u.cm / u.s / u.keV,\n",
-    "                                 reference = 100 * u.keV)\n",
+    "index = -3\n",
+    "K = 10**-3 / u.cm / u.cm / u.s / u.keV,\n",
+    "piv = 100 * u.keV\n",
+    "\n",
+    "spectrum = Powerlaw()\n",
```
[the remark this answers is not in the document]
Comment (eneights) [reply docs/tutorials/DetectorResponse.ipynb:642]: This is fixed in the latest commit.

`spectrum.K = K` does not work

File: cosipy/response/FullDetectorResponse.py [hunk 2]
```diff
@@ -364,15 +364,16 @@ def get_expectation():
             psr = response.get_point_source_response(exposure_map)
             
             # Spectrum
-            spectrum_module = importlib.import_module('gammapy.modeling.models')
-            spectrum_class = getattr(spectrum_module, config['source:spectrum:class'])
-            spectrum = spectrum_class(*config.get('source:spectrum:args', []),
-                                      **config.get('source:spectrum:kwargs', {}))
-
-            logger.info(f"Using spectrum:\n {spectrum}")
+            spectrum_module = importlib.import_module('astromodels.core.model_parser')
```
[the remark this answers is not in the document]
Comment (eneights) [reply cosipy/response/FullDetectorResponse.py]: This has been fixed

File: cosipy/response/FullDetectorResponse.py [hunk 3]
```diff
@@ -364,15 +364,16 @@ def get_expectation():
...
+            model_load = getattr(spectrum_module, 'ModelParser')
+            model = model_load(model_dict=config['sources']).get_model()
+            
+            for src_name, src in model.point_sources.items():
+                for comp_name, component in src.components.items():
+                    logger.info(f"Using spectrum:\n {component.shape}")
             
             # Expectation
-            expectation = psr.get_expectation(spectrum).project('Em')
+            expectation = psr.get_expectation(spectrum).project('Em') ######
```
[the remark this answers is not in the document]
Comment (eneights) [reply cosipy/response/FullDetectorResponse.py]: Whoops, didn't mean to leave that in there. Fixed!

File: cosipy/response/PointSourceResponse.py [hunk 4]
```diff
@@ -4,6 +4,10 @@
 
 from astropy.units import Quantity
 
+from scipy import integrate
+
+from threeML import *
```
[the remark this answers is not in the document]
Comment (eneights) [reply cosipy/response/PointSourceResponse.py]: Updated to import each model

File: cosipy/response/PointSourceResponse.py [hunk 5]
```diff
@@ -56,7 +60,23 @@ def get_expectation(self, spectrum):
         
         eaxis = self.photon_energy_axis
         
-        flux = Quantity([spectrum.integral(lo_lim, hi_lim)
+        if (isinstance(spectrum, Blackbody) or isinstance(spectrum, ModifiedBlackbody) or isinstance(spectrum, NonDissipativePhotosphere) or
```
[the remark this answers is not in the document]
Comment (eneights) [reply cosipy/response/PointSourceResponse.py:63]: Didn't change this in the latest commit. Will plan to change this before incorporating the integration into the 3ML code

File: cosipy/response/PointSourceResponse.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (israelmcmc) [reply cosipy/response/PointSourceResponse.py]: Thanks. This should change again after we modify 3ML to do the integration.

File: cosipy/response/PointSourceResponse.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (israelmcmc) [reply cosipy/response/PointSourceResponse.py:63]: Ok. I'll open an issue to remember this.

File: .DS_Store
[the remark this answers is not in the document]
Comment (israelmcmc) [reply .DS_Store:1]: Hmmm most are still there in the "Files changed" tab 🙃 . I'll PM you.

File: cosipy/response/PointSourceResponse.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (israelmcmc) [reply cosipy/response/PointSourceResponse.py:63]: Sounds good. I'll open an issue about this to remember us

File: .DS_Store
[the remark this answers is not in the document]
Comment (eneights) [reply .DS_Store:1]: They should actually be gone now!

**Comment by eneights** [comment]
I moved the COSILike class to cosipy.threeml and added an example notebook for the spectral fit

**Pull request merged**